### PR TITLE
Feature/102 call to action

### LINKED ISSUE #97: Index page: Don't show actions bar if edit/delete have been disabled
Also, change colspan in empty body

### LINKED ISSUE #102: Refactor: Index: Change the no results message to include a call to action
EG: "There are no results. You can start by creating [a new object here](some_link)"

Also - move it out of the table view

Also - move this into own class so it can be tested as a unit.

### PR DESCRIPTION
**Pull request by jordanmaguire**
- Fix #97: Only show actions if edit/delete are enabled
- Fix #102 Add an empty message and a call to action to the index

screen shot 2016-03-12 at 2 43 25 pm

Time to get drunk now

### CODE REVIEW

File: lib/frontier/views/index/empty_message_and_call_to_action.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+private
+
+  def empty_message_with_call_to_action
+    raw = <<-STRING
+%p
+  #{empty_message_without_call_to_action}
+  - if #{Frontier::Authorization::Check.new(model_configuration, model_configuration.as_constant, :new)}
+    = link_to(\"Click here to create one.\", #{model_configuration.url_builder.new_path})
```
Comment (pzi) [inline lib/frontier/views/index/empty_message_and_call_to_action.rb]: No need for `Click here`, just link `create one`.

**Comment by pzi** [comment]
Not a big fan of having the main action on an empty page pushed out all the way to the right

**Comment by jordanmaguire** [comment]
> Not a big fan of having the main action on an empty page pushed out all the way to the right

Got a suggestion for an alternative?

File: lib/frontier/views/index/empty_message_and_call_to_action.rb (hunk 1 above)
Comment (pzi) [inline lib/frontier/views/index/empty_message_and_call_to_action.rb]: Or better, `Create a/n %{label}`.

Comment (jordanmaguire) [inline lib/frontier/views/index/empty_message_and_call_to_action.rb]: Feels repetitive as is.

> There are no users. <a>Create a user</a>.

vs

> There are no users. <a>Create one now</a>.

Still think we need the noun two times?

**Comment by pzi** [comment]
If there is none, the following could be on a page:

```haml
%h2 Work Site Management
%p There are no work sites.
%p= link_to "Add New Work Site", some_path, class: "btn"
```

Only if there are none yet. So you have a brief explanation of what's going on and then a clear call to action button to create one.

File: lib/frontier/views/index/empty_message_and_call_to_action.rb (hunk 1 above)
Comment (jordanmaguire) [inline lib/frontier/views/index/empty_message_and_call_to_action.rb]: Have gone with

`There are no users. <a>Create a user now</a>`

Comment (pzi) [inline lib/frontier/views/index/empty_message_and_call_to_action.rb]: Hence [my suggestion](https://github.com/thefrontiergroup/frontier_generators/pull/106#issuecomment-196101745) to drop the link from the paragraph and use the main call to action button underneath it instead.

**Pull request merged**